### PR TITLE
Upgrade FAB to 4.4.1

### PR DESCRIPTION
**Pull request by potiuk**
This upgrades FAB to version 4.4.1 and brings all the changes done to security manager between v4.3.11 and v4.4.1. As usual we need to manually synchronize the changes to security manager which we override in FAB provider.

This one also has the nice effect of bumping limits for a number of dependencies we have, such as Werkzeug (finally <4 rather than <3) and Flask (<3 rather than <2.3).

Still SQLAlchemy is not bumped though.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
I'd really appreciate verifying if I applied all the changes between v4.3.11 and v4.4.1 that we should incorporate in our override @vincbeck especially - but also others.

File: tests/www/views/test_session.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def test_check_active_user(app, user_client):
     user.active = False
     resp = user_client.get("/home")
     assert resp.status_code == 302
-    assert "/logout" in resp.headers.get("Location")
+    assert "/login/?next=http%3A%2F%2Flocalhost%2Fhome" in resp.headers.get("Location")
```
Comment (potiuk) [inline tests/www/views/test_session.py:100]: This is result of https://github.com/dpgaspar/Flask-AppBuilder/pull/2192 from FAB. And I think it's a better behaviour, because it will return the user back to the page they were originally in case someone re-activates them after they've been logged out.

File: airflow/api_connexion/endpoints/pool_endpoint.py [hunk 2]
```diff
@@ -114,7 +114,10 @@ def patch_pool(
         update_mask = [i.strip() for i in update_mask]
         _patch_body = {}
         try:
-            update_mask = [
+            # MyPy infers a List[Optional[str]]  type here but it should be a List[str]
```
Comment (potiuk) [inline airflow/api_connexion/endpoints/pool_endpoint.py:117]: Not sure why this one started to happen (mypy quirks) but the error signalled by mypy here makes no sense.

File: airflow/api_connexion/endpoints/pool_endpoint.py [hunk 3]
```diff
@@ -114,8 +114,11 @@ def patch_pool(
         update_mask = [i.strip() for i in update_mask]
         _patch_body = {}
         try:
+            # MyPy infers a List[Optional[str]]  type here but it should be a List[str]
+            # there is no way field is None here (UpdateMask is a List[str])
```
Comment (vincbeck) [inline airflow/api_connexion/endpoints/pool_endpoint.py:118]:
```suggestion
            # there is no way the field is None here (update_mask is a List[str])
```

File: scripts/in_container/install_airflow_and_providers.py
Comment (vincbeck) [inline scripts/in_container/install_airflow_and_providers.py:1]: Is this change related to FAB upgrade? It seems it is not

Comment (potiuk) [reply scripts/in_container/install_airflow_and_providers.py:1]: well. Mypy started to complain in this PR. I think it's the usual "never know when MyPy starts complaining about something". It does not seem to be related to any other package upgraded in this PR.

File: airflow/api_connexion/endpoints/pool_endpoint.py (hunk 3 above)
Comment (potiuk) [reply airflow/api_connexion/endpoints/pool_endpoint.py:118]: Actually UpdateMask is type of update_mask (see few lines above) so it's good description (But I might add it to be more explicit once it passes all tests)

**Comment by vincbeck** [comment]
I checked [the diff between FAB 4.3.11 and 4.4.1](https://github.com/dpgaspar/Flask-AppBuilder/compare/release/4.3.11...release/4.4.1) and I dont think you forgot anything 👌

File: scripts/in_container/install_airflow_and_providers.py
Comment (vincbeck) [reply scripts/in_container/install_airflow_and_providers.py:1]: Oh so it is related 👌 You can ignore my comment then

**Review by vincbeck — APPROVED** [review]

File: scripts/in_container/install_airflow_and_providers.py
Comment (potiuk) [reply scripts/in_container/install_airflow_and_providers.py:1]: SOMEWHAT related :)
